src/components/DataTypes/array.tsx

```diff
@@ -190,25 +143,114 @@ export const getFilter = (schema: JSONSchema, value: string) => {
 		return {
 			minItems: 1,
 			items: {
-				properties: Object.keys(schema.items.properties).reduce(
-					(props: NonNullable<JSONSchema['properties']>, propKey: string) => {
-						props[propKey] = {
-							// @ts-expect-error
-							regexp: {
-								pattern: regexEscape(value),
-								flags: 'i',
+				anyOf: [
+					...Object.entries(schema.items.properties).reduce(
+						(props: any[], [propKey, propValue]: [string, JSONSchema]) => {
+							const subSchema = isFullTextSearch
+								? { ...propValue, oneOf: undefined, enum: undefined }
+								: propValue;
+							const model = getDataModel(subSchema);
+							const operators = model?.operators(schema);
+							const newOperator = convertOperator(operator, operators);
+							const filter = model?.createFilter(
+								key,
+								propKey,
+								newOperator,
+								value,
+								subSchema,
+								true,
+							);
+							props.push(filter);
+							return props;
+						},
+						[],
+					),
+				],
+			},
+		};
+	} else if (
+		!!schema?.items &&
+		typeof schema.items !== 'boolean' &&
+		'type' in schema.items
+	) {
+		const filter = getModelFilter(
+			schema.items,
+			key,
+			field,
+			operator,
+			value,
+			isFullTextSearch,
+			convertOperator,
+		);
```

Comment:
The difference between this filter and the one below is just `schema` vs `schema.items` right? Maybe we can have just 1 `const filter` above/outside this if and just determine which to use with the same if?